src/Mappers/InventoryMapper.php

```diff
@@ -110,19 +111,50 @@ public function createInventoryDTOsFromVariation($variationData)
         continue;
       }
 
+      // Avl Immediately. Net Stock.
+      $onHand = isset($stock['netStock']) ? $stock['netStock'] : null;
+      // Already Ordered items.
+      $onOrder = isset($stock['reservedStock']) ? $stock['reservedStock'] : null;
+
+      // key for a potential preexisting DTO that we need to merge with
+      $dtoKey = $supplierId . '_' . $supplierPartNumber;
+
+      /** @var RequestDTO $existingDTO */
+      $existingDTO = $inventoryDTOs[$dtoKey];
+
+      if (isset($existingDTO))
+      {
+        // merge with previous values for this suID
+        if (isset($existingDTO->getQuantityOnHand()))
+        {
+          $onHand += $existingDTO->getQuantityOnHand();
+        }
+
+        if (isset($existingDTO->getQuantityOnOrder()))
+        {
+          $onOrder += $existingDTO;
+        }
+
+        if (!isset($nextAvailableDate))
+        {
+          $nextAvailableDate = $existingDTO->getItemNextAvailabilityDate();
```

Comment:
For next available date should we take whichever date is earliest?

Reply:
I originally felt that way as well, but the next available date is coming from the variation level (outside of this loop), not the individual warehouse.

Reply:
so actually yeah this code doesn't do anything in the first place. whoops.